pkg/transport/process/pipeline.go

```diff
@@ -0,0 +1,83 @@
+package process
+
+import (
+	"context"
+	"os"
+
+	"archive/tar"
+	"fmt"
+	"io/ioutil"
+
+	cdv2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+)
+
+type resourceProcessingPipelineImpl struct {
+	processors []ResourceStreamProcessor
+}
+
+func (p *resourceProcessingPipelineImpl) Process(ctx context.Context, cd cdv2.ComponentDescriptor, res cdv2.Resource) (*cdv2.ComponentDescriptor, cdv2.Resource, error) {
```

Comment:
exported functions should have a docstring

Reply:
i skipped the comment here because everything one needs to know is already documented in the interface declaration

Reply:
ok but then we should have a `docs.go` that describes what the processor is doing.
Because the interface is known but the internals and the interface for the called processes